src/legacy/core_plugins/vis_default_editor/public/components/controls/time_interval.tsx

```diff
@@ -25,6 +25,31 @@ import { FormattedMessage } from '@kbn/i18n/react';
 
 import { isValidInterval, AggParamOption } from '../../legacy_imports';
 import { AggParamEditorProps } from '../agg_param_props';
+import { search } from '../../../../../../plugins/data/public';
+const { parseEsInterval, InvalidEsCalendarIntervalError } = search.aggs;
+
+function getInvalidEsMessage(interval: string) {
+  return i18n.translate(
+    'visDefaultEditor.controls.timeInterval.invalidCalendarIntervalErrorMessage',
+    {
+      defaultMessage: 'Invalid calendar interval: {interval}, value must be 1',
+      values: { interval },
+    }
+  );
+}
+
+function isValidEsInterval(interval: string) {
+  try {
+    parseEsInterval(interval);
+    return true;
+  } catch (e) {
+    if (e instanceof InvalidEsCalendarIntervalError) {
+      return false;
+    }
+
+    return true;
+  }
+}
```

Comment:
That function already exists on `search.aggs`, so we can just use `isValidEsInterval` directly instead of `parseEsInterval`.

Reply:
yes, but that function catches all exceptions, and in this case I wanted to catch particular `InvalidEsCalendarIntervalError`.

Reply:
Okay I see. Maybe it would make sense to rename it to `isValidCalendarInterval` to make the purpose a bit clearer?

Reply:
Agree, it's clearer.
Updated.